alignedbuff/alignedbuff_test.go

```diff
@@ -103,7 +103,15 @@ func TestAlignedBuff32(t *testing.T) {
 
 	b := NewWithData(b0.data)
 
-	if len(b0.Data()) != 4*4 {
+	// Sigh. The Linux kernel expects certain nftables payloads to be padded to
+	// the uint64 next alignment. Now, on 64bit platforms this will be a 64bit
```

Comment:
Doesn’t the Linux kernel expect payloads to be padded to **`int`** alignment (not uint64)? The size of an `int` differs between 32-bit and 64-bit as you outlined.

Reply:
I (re) checked with x_tables.h (https://elixir.bootlin.com/linux/v5.17.7/source/include/uapi/linux/netfilter/x_tables.h#L93) and added that documentation reference to the package comment of `alignedbuff` so others can quickly find and check my reasoning. The alignment is done based on the alignment of a struct consisting of unsigned ints of 8, 16, 32, 64bits. I cross-checked with a tiny C program on amd64 as well as i386 and it gave me the same alignment results as the Go compiler on each architecture respectively: 64bits and 32bits alignments.

In consequence, my calculation of the expected marshalled data size was incorrect for 32bit platforms and this patch fixes my original incorrect expectation. The implemented alignment itself was and still is correct.